src/py/flwr/simulation/run_simulation.py

```diff
@@ -17,92 +17,185 @@
 import argparse
 import asyncio
 import json
+import logging
 import threading
 import traceback
-from logging import ERROR, INFO, WARNING
+from logging import DEBUG, ERROR, INFO, WARNING
 from time import sleep
-from typing import Any, Callable
+from typing import Dict, Optional
 
 import grpc
 
+from flwr.client import ClientApp
 from flwr.common import EventType, event, log
+from flwr.common.typing import ConfigsRecordValues
 from flwr.server.driver.driver import Driver
 from flwr.server.run_serverapp import run
+from flwr.server.server_app import ServerApp
 from flwr.server.superlink.driver.driver_grpc import run_driver_api_grpc
 from flwr.server.superlink.fleet import vce
 from flwr.server.superlink.state import StateFactory
-from flwr.simulation.ray_transport.utils import enable_tf_gpu_growth
+from flwr.simulation.ray_transport.utils import (
+    enable_tf_gpu_growth as enable_gpu_growth,
+)
 
 
+# Entry point from CLI
+def run_simulation_from_cli() -> None:
+    """Run Simulation Engine from the CLI."""
+    args = _parse_args_run_simulation().parse_args()
+
+    # Load JSON config
+    backend_config_dict = json.loads(args.backend_config)
+
+    _run_simulation(
+        server_app_attr=args.server_app,
+        client_app_attr=args.client_app,
+        num_supernodes=args.num_supernodes,
+        backend_name=args.backend,
+        backend_config=backend_config_dict,
+        working_dir=args.dir,
+        driver_api_address=args.driver_api_address,
+        enable_tf_gpu_growth=args.enable_tf_gpu_growth,
+        verbose_logging=args.verbose,
+    )
+
+
+# Entry point from Python session (script or notebook)
+# pylint: disable=too-many-arguments
+def run_simulation(
+    server_app: ServerApp,
+    client_app: ClientApp,
+    num_supernodes: int,
+    backend_name: str = "ray",
+    backend_config: Optional[Dict[str, ConfigsRecordValues]] = None,
+    enable_tf_gpu_growth: bool = False,
+    verbose_logging: bool = False,
+) -> None:
+    r"""Run a Flower App using the Simulation Engine.
+
+    Parameters
+    ----------
+    server_app : ServerApp
+        The `ServerApp` to be executed. It will send messages to different `ClientApp`
+        instances.
```

Comment:
```suggestion
        The `ServerApp` to be executed. It will send messages to different `ClientApp`
        instances running on different (virtual) SuperNodes.
```